docs/source/gpu.rst

```diff
@@ -0,0 +1,164 @@
+
+If you have cudf installed then you should be able to convert a Pandas-backed
+Dask DataFrame to a cuDF-backed Dask DataFrame as follows:
+
+.. code-block:: python
+
+   import cudf
+
+   df = df.map_partitions(cudf.from_pandas)  # convert pandas partitions into cudf partitions
+
+However, cuDF does not support the entire Pandas interface, and so a variety of
+Dask DataFrame operations will not function properly.
+
+
+Arrays
+~~~~~~
+
+.. note:: Dask's integration with CuPy relies on features recently added to
+   NumPy and CuPy, particularly in version ``numpy>=1.17`` and ``cupy>=6``
+
+`Chainer's CuPy <https://cupy.chainer.org/>`_ library provides a GPU
+accelerated NumPy-like library that interoperates nicely with Dask Array.
+
+If you have CuPy installed then you should be able to convert a NumPy-backed
+Dask Array into a CuPy backed Dask Array as follows:
+
+.. code-block:: python
+
+   import cupy
+
+   x = x.map_blocks(cupy.asarray)
+
+CuPy is fairly mature and adheres closely to the NumPy API.  However, small
+differences do exist and these can cause Dask Array operations to function
+improperly. Check the
+`CuPy Reference Manual <https://docs-cupy.chainer.org/en/stable/reference/index.html>`_
+for API compatibility.
+
+
+Scikit-Learn
+~~~~~~~~~~~~
+
+There are a variety of GPU accelerated machine learning libraries that follow
+the Scikit-Learn Estimator API of fit, transform, and predict.  These can
+generally be used within `Dask-ML's <https://ml.dask.org>`_ meta estimators,
+such as `hyper parameter optimization <https://ml.dask.org/hyper-parameter-search.html>`_.
+
+Some of these include:
+
+-  `Skorch <https://skorch.readthedocs.io/>`_
+-  `cuML <https://rapidsai.github.io/projects/cuml/en/latest/>`_
+-  `LightGBM <https://github.com/Microsoft/LightGBM>`_
+-  `XGBoost <https://xgboost.readthedocs.io/en/latest/>`_
+-  `Thunder SVM <https://github.com/Xtra-Computing/thundersvm>`_
+-  `Thunder GBM <https://github.com/Xtra-Computing/thundergbm>`_
+
+
+Setup
+-----
+
+From the examples above we can see that the user experience of using Dask with
+GPU-backed libraries isn't very different from using it with CPU-backed
+libraries.  However, there are some changes you might consider making when
+setting up your cluster.
+
+Restricting Work
+~~~~~~~~~~~~~~~~
+
+By default Dask allows as many tasks as you have CPU cores to run concurrently.
+However if your tasks primarily use a GPU then you probably want far fewer
+tasks running at once.  There are a few ways to limit parallelism here:
+
+-   Limit the number of threads explicitly on your workers using the
+    ``--nthreads`` keyword in the CLI or the ``ncores=`` keyword the
+    Cluster constructor.
+-   Use `worker resources <https://distributed.dask.org/en/latest/resources.html>`_ and tag certain
+    tasks as GPU tasks so that the scheduler and limit them, while leaving the
+    rest of your CPU cores for other work
+
+Specifying GPUs per Machine
```

Comment:
I haven't exposed this in `dask-yarn` yet, but skein supports Hadoop 3 GPU resource requests (https://jcrist.github.io/skein/specification.html#resources), which set these environment variables automatically. If I was to expose this in `dask-yarn`, perhaps adding a link to that section from here would be appropriate?

Reply:
Yes, that sounds good.  We should do something similar with dask-kubernetes